common/services/user.test.js

```diff
@@ -93,6 +93,86 @@ describe('User service', function () {
     })
   })
 
+  describe('#getSupplierId()', function () {
+    let tokenData, result
+
+    context('when supplier exists', function () {
+      beforeEach(async function () {
+        nock(configStub.AUTH_PROVIDERS.hmpps.groups_url('test'))
+          .get('/')
+          .reply(200, JSON.stringify(authGroups))
+
+        tokenData = { authorities: ['ROLE_PECS_SUPPLIER'] }
+
+        result = await getSupplierId(encodeToken(tokenData))
+      })
+
+      it('requests the user’s groups from HMPPS SSO', function () {
+        expect(nock.isDone()).to.be.true
+      })
+
+      it('calls reference with supplier key', function () {
+        expect(referenceDataStub.getSupplierByKey).to.be.calledWithExactly(
+          supplierStub.key
+        )
+      })
+
+      it('returns an Array of location objects', function () {
+        expect(result).to.deep.equal(supplierStub.id)
+      })
+    })
+
+    context('when supplier does not exist', function () {
```

Comment:
Added some extra tests to cover the other cases when there is either no supplier role or no auth groups are returned.